### PR TITLE
Corda 5 Glossary Functions and Styling

### PR DESCRIPTION
**Pull request by nadinequinr3**
Test page to review the functions in action - https://nadine.preview.docs.r3.com/en/platform/corda/5.0-beta/introduction/glossary.html

(The content labelled "TEST" at the end of the page is only temporary to help to review the PR. These tooltips would actually appear throughout the Corda 5 content on any page. I've only added a few items to the glossary list, for testing purposes. This list will need to be fleshed out over the coming weeks.)

### CODE REVIEW

**Comment by AnnaK-R3** [comment]
There's no punctuation marks between terms and description and formatting looks a bit weird. If we want to keep the unordered list, I'd add a colon or an M dash between the term and its definition, if possible. Another option is to use a definition list which imo makes more sense when it comes to structuring a glossary. See https://www.oxygenxml.com/dita/styleguide/Syntax_and_Markup/c_Definition_Lists_vs_Tables.html

**Comment by nadinequinr3** [comment]
> There's no punctuation marks between terms and descriptions and the formatting looks a bit weird. If we want to keep the unordered list, I'd add a colon or an M dash between the term and its definition, if possible. Another option is to use a definition list which imo makes more sense when it comes to structuring a glossary. See https://www.oxygenxml.com/dita/styleguide/Syntax_and_Markup/c_Definition_Lists_vs_Tables.html

Thanks, Anna. I had played with using a table but was trying to avoid them after our experience with the Release Notes. I hadn't thought of a definition list though so will test that now.

**Review by Niamh25 — APPROVED** [review]

**Comment by nadinequinr3** [comment]
Merging the following:

-  shortcode to retrieve tootlip definitions for inline terms
-  shortcode to retrieve full list of terms and definitions
- css for tooltips
- css for glossary page
- "draft" glossary page - leaving as draft to exclude from builds until we have a more complete and reviewed list of terms (to come in future PR(s))
- JSON file to define Corda 5 terms

File: themes/doks/layouts/shortcodes/definition.html
Comment (yarivus) [inline themes/doks/layouts/shortcodes/definition.html:1]: Nice one 🥇

File: themes/doks/layouts/shortcodes/glossary.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{{/* Function to display the full list of terms and definitions defined for Corda 5 in glossaryitems.json. */}}
+{{/*The "definition" field in glossaryitems.json should only be one sentence. This is the text displayed as the ToolTip throughout the content and also displayed by default on the Glossary page.*/}}
+{{/*The "more" field in glossaryitems.json displays any additional content displayed by expanding the paragraph on the Glossary page.*/}}
+
+{{ $glossarylist := getJSON "content/en/platform/corda/5.0-beta/glossaryitems.json" }}
```
Comment (yarivus) [inline themes/doks/layouts/shortcodes/glossary.html:5]: I think that the next step is to generalize it.
In other words not be version specific ...probably the glossaryitems for Corda haven't changed in a while and won't change much going fwd ...
Then you'd have a master `glossaryitems.json` file and that would suit all versions of Corda.
@nadinequinr3 / @colin-hillis

**Pull request merged**